node/src/reactor/joiner.rs

```diff
@@ -762,14 +762,19 @@ impl reactor::Reactor for Reactor {
                 self.consensus.handle_event(effect_builder, rng, event),
             ),
             Event::ConsensusAnnouncement(announcement) => match announcement {
-                ConsensusAnnouncement::Handled(block_header) => reactor::wrap_effects(
-                    Event::LinearChainSync,
-                    self.linear_chain_sync.handle_event(
-                        effect_builder,
-                        rng,
+                ConsensusAnnouncement::Handled(block_header) => {
+                    let mut effects = Effects::new();
+                    if block_header.switch_block() {
+                        let reactor_event =
+                            Event::ChainspecLoader(chainspec_loader::Event::CheckForNextUpgrade);
+                        effects.extend(self.dispatch_event(effect_builder, rng, reactor_event));
```

Comment:
Could this be done in `linear_chain_sync` instead of the reactor itself?

Reply:
I have just removed it as that's wrong :grimacing:

Reply:
We shouldn't be checking for the next upgrade on switch blocks but check if we should shut down on switch block! And that's going to happen in the `linear_chain_sync`.